packages/okta-adapter/src/dependency_changers/order_policies_by_priority.ts

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright 2024 Salto Labs Ltd.
+ * Licensed under the Salto Terms of Use (the "License");
+ * You may not use this file except in compliance with the License.  You may obtain a copy of the License at https://www.salto.io/terms-of-use
+ *
+ * CERTAIN THIRD PARTY SOFTWARE MAY BE CONTAINED IN PORTIONS OF THE SOFTWARE. See NOTICE FILE AT https://github.com/salto-io/salto/blob/main/NOTICES
+ */
+
+import _ from 'lodash'
+import {
+  Change,
+  DependencyChange,
+  DependencyChanger,
+  InstanceElement,
+  ReferenceExpression,
+  dependencyChange,
+  getChangeData,
+  isAdditionChange,
+  isInstanceChange,
+  isReferenceExpression,
+} from '@salto-io/adapter-api'
+import { deployment } from '@salto-io/adapter-components'
+import { getParent } from '@salto-io/adapter-utils'
+import { logger } from '@salto-io/logging'
+import { POLICY_PRIORITY_TYPE_NAMES, POLICY_RULE_PRIORITY_TYPE_NAMES } from '../constants'
+import { ALL_SUPPORTED_POLICY_NAMES, POLICY_RULE_TYPES_WITH_PRIORITY_INSTANCE } from '../filters/policy_priority'
+
+const log = logger(module)
+type ChangeWithKey = deployment.dependency.ChangeWithKey<Change<InstanceElement>>
+
+const createDependencyChange = (prev: ChangeWithKey, current: ChangeWithKey): DependencyChange[] => [
+  dependencyChange('add', current.key, prev.key),
+]
+
+const groupChangesByType = (changes: ChangeWithKey[]): Record<string, ChangeWithKey[]> =>
+  changes.reduce(
+    (acc, change) => {
+      const instance = getChangeData(change.change)
+      const { typeName } = instance.elemID
+      if (!acc[typeName]) {
+        acc[typeName] = []
+      }
+      acc[typeName].push(change)
+      return acc
+    },
+    {} as Record<string, ChangeWithKey[]>,
+  )
```

Comment:
just because we had this discussion - this IMO is a case where there is not a real reason to use `reduce` and `_.groupBy` would have been clearer
(same applies for the other `reduce`, though you can first filter for changes with parent using `hasValidParent`)

WDYT ?
```suggestion
const groupChangesByType = (changes: ChangeWithKey[]): Record<string, ChangeWithKey[]> =>
  _.groupBy(
    changes,
    change => getChangeData(change.change).elemID.typeName
  )
```